common/src/main/java/org/valkyrienskies/mod/mixin/entity/MixinEntity.java

```diff
@@ -218,9 +217,9 @@ private void preCalculateViewVector(final float xRot, final float yRot, final Ca
     @Inject(method = "shouldRender", at = @At("HEAD"), cancellable = true)
     private void onShouldRender(double d, double e, double f, CallbackInfoReturnable<Boolean> cir) {
         if (this.draggingInformation.isEntityBeingDraggedByAShip() && this.level.isClientSide) {
-            final ClientShip ship = VSGameUtilsKt.getShipObjectWorld((ClientLevel) this.level).getAllShips().getById(this.draggingInformation.getLastShipStoodOn());
-            if (ship != null) {
-                final ShipTransform shipTransform = ship.getRenderTransform();
+            final Ship ship = VSGameUtilsKt.getShipObjectWorld(this.level).getAllShips().getById(this.draggingInformation.getLastShipStoodOn());
+            if (ship != null && ship instanceof final ClientShip clientShip) {
+                final ShipTransform shipTransform = clientShip.getRenderTransform();
```

Comment:
I just noticed, if anything invoked `shouldRender` on server side, they'll also expect it to work correctly. So maybe do a check here:

```suggestion
            if (ship != null) {
                final ShipTransform shipTransform = ship instanceof final ClientShip clientShip ? clientShip.getRenderTransform() : ship.getTransform();
```